### ISSUE TITLE
Issues accessing pathlib.Path.parents by index in Python 3.14

### ISSUE THREAD

**Issue by felixfontein** [pylint-dev/pylint#10681]
### Bug description

```python
On Python 3.14 with pylint 4.0.1, process this Python file:

# pylint: disable=missing-module-docstring
from pathlib import Path

cwd = Path.cwd()
parents = cwd.parents
print(parents[0].name)

This produces:

t.py:6:6: E1101: Instance of 'tuple' has no 'name' member (no-member)

This does not happen with Python 3.13.

(Related issue in the past: #5832)
```

### Configuration

```ini

```

### Command used

```shell
pylint t.py
```

### Pylint output

```python
************* Module t
t.py:6:6: E1101: Instance of 'tuple' has no 'name' member (no-member)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

Pylint passes without reporting errors.

### Pylint version

```shell
pylint 4.0.1
astroid 4.0.1
Python 3.14.0 (main, Oct  8 2025, 21:26:41) [GCC 14.2.0]
```

### OS / Environment

`python:3.14-slim` Docker container with `pip install pylint`

### Additional dependencies

```python

```
